addons/flannel/0.21.5/install.sh

```diff
@@ -316,11 +316,27 @@ function remove_weave() {
     kubectl -n kube-system delete serviceaccount weave-net
     kubectl -n kube-system delete secret weave-passwd
 
-    # all nodes
-    ip link delete weave
-    rm -rf /var/lib/weave
-    rm -rf /etc/cni/net.d/*weave*
-    rm -rf /opt/cni/bin/weave*
+    # It should be executed in all nodes either
+    # See that the task used to run in the nodes also has this commands
+    # Delete the weave network interface, if it exists
+    if ip link show weave > /dev/null 2>&1; then
+        ip link delete weave
+    fi
+
+    # Delete the /var/lib/weave directory, if it exists
```

Comment:
this one here may be safe to not need an if conditional, but doesn't hurt for safety (pretty sure `rm -rf` will always succeed? at least if there's an issue with `/var/lib` though it will just move on... theoretically)

Reply:
Done